### PR TITLE
Feat: add warning lint for object mutation

### PR DESCRIPTION
**Pull request by jakedaleweb**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,6 +22,7 @@ module.exports = {
     "plugin:prettier/recommended",
   ],
   rules: {
+    "functional/immutable-data": "warn",
```
Comment (jakedaleweb) [inline index.js:25]: start with a warning, I have a feeling there will be a bit to work through

Comment (jpdenford) [reply index.js:25]: Can probably also add some of the other options like 
```
ignoreClass
ignoreImmediateMutation
```
which might reduce some noise.

Comment (jakedaleweb) [reply index.js]: @jpdenford good call, have done

**Review by jpdenford — APPROVED** [review]
Noice

**Pull request merged**